internal/tenant/tenant.go

```diff
@@ -62,3 +63,38 @@ func (t tenantService) GetTenantInfoByNamespace(apiURL string, ns string) (InfoL
 
 	return tenantInfo, nil
 }
+
+// returns true if the cluster the ns is on has reached maximum capacity
+func (t tenantService) HasReachedMaxCapacity(apiURL, ns string) (bool, error) {
+
+	ti, err := t.GetTenantInfoByNamespace(apiURL, ns)
+	if err != nil {
+		return true, err
+	}
+
+	if len(ti.Errors) != 0 {
+		firstError := ti.Errors[0]
+		err = fmt.Errorf("%s - %s", firstError.Code, firstError.Detail)
+		return true, err
+	}
+
+	namespaces := ti.Data[0].Attributes.Namespaces
```

Comment:
should we check `if ti.Data != nil` or/and `if len(ti.Data) > 0` before accessing the first element to avoid the nil pointer exception

Reply:
hmm, good catch but my assumption was that the json is expected to return data if there aren't errors
However, good idea to check that assumption :) and sufficient to check the len

```go
	var x []int
	fmt.Printf("%d", len(x)) // print 0
```